File: leetcode/1431.cpp
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Kids With the Greatest Number of Candies.
// Memory Usage: 8.9 MB, less than 58.99% of C++ online submissions for Kids With the Greatest Number of Candies.
class Solution {
 public:
  vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies) {
    auto m = max_element(candies.begin(), candies.end());
    vector<bool> res = {};
    for (auto c : candies) {
      res.push_back(c + extraCandies >= *m ? true : false);
    }
    return res;
  }
};

File: leetcode/1431.py
"""
Runtime: 40 ms, faster than 68.52% of Python3 online submissions for Kids With the Greatest Number of Candies.
Memory Usage: 14.2 MB, less than 80.38% of Python3 online submissions for Kids With the Greatest Number of Candies.
"""


class Solution:
    def kidsWithCandies(
        self, candies: List[int], extraCandies: int
    ) -> List[bool]:
        # Normalize so largest element is 0
        m = max(candies)
        normalized = [c - m for c in candies]

        # add extras to each, and they have the most if num is positive
        return [n + extraCandies >= 0 for n in normalized]
